internal/provider/resources/routing_rules.go

```diff
@@ -262,23 +272,47 @@ func (r *RoutingRules) updateState(ctx context.Context, diags *diag.Diagnostics,
 	data.Rule = latest.Workflow.Rule
 	data.Version = types.StringValue(*latest.Workflow.Version)
 
+	tflog.Debug(ctx, fmt.Sprintf("Updating state to: %+v", data))
+
 	// Save updated data into Terraform state
 	diags.Append(state.Set(ctx, data)...)
 }
 
-// Posts a new routing-rules version to P0. This is used for create, update, and delete. Note that delete does not delete, but
-// rather posts a default routing-rules set.
+// Posts a new routing-rules version to P0. This is used for create, update, and delete.
+// Note that delete does not delete, but rather posts a default routing-rules set.
 func (r *RoutingRules) postVersion(ctx context.Context, data RoutingRulesModel, diag *diag.Diagnostics, state *tfsdk.State) {
-	workflowUpdate := WorkflowLatestApi{Workflow: RoutingRulesRepr{Rule: data.Rule}}
+	tflog.Debug(ctx, fmt.Sprintf("Update Data: %+v", data))
 
-	var latest WorkflowLatestApi
-	_, postErr := r.data.Post("workflow", &workflowUpdate, &latest)
+	var current RoutingRulesModel
+	diag.Append(state.Get(ctx, &current)...)
+	if diag.HasError() {
+		return
+	}
+
+	tflog.Debug(ctx, fmt.Sprintf("Current workflow state: %+v", current))
+
+	var currentVersionPtr *string
+
+	if !current.Version.IsUnknown() && !current.Version.IsNull() {
+		currentVersion := current.Version.ValueString()
+		tflog.Debug(ctx, fmt.Sprintf("Current version: %s", currentVersion))
+		currentVersionPtr = &currentVersion
```

Comment:
Doesn't this mean that if someone changes the version manually on the UI, the terraform state will be invalid? What happens if you delete your current Tf state and start from scratch?

Reply:
If the state is changed in the UI, it will work. Terraform does a read as part of the plan. It will show the differences and ask to apply them.

What this does is prevent Terraform from making changes if the state is concurrently modified after the plan is created.

If the TF state is deleted, it needs to be imported first. But that is existing behavior, not specific to this PR.